rfc/graph-feature-flags.md

```diff
@@ -0,0 +1,307 @@
+
+```bash
+npx wgc subgraph publish products --schema ../demo/subgraphs/products/products.graphql --routing-url http://localhost:4001/graphql --label=team=A
+```
+
+Let's say, we'd like to use the `@override` directive that overrides a field from another Subgraph,
+and we want to gradually roll out this new feature to our users.
+
+We can add a `--feature-flag` flag to the `subgraph publish` command:
+
+```bash
+npx wgc subgraph publish products --schema ../demo/subgraphs/products/products.graphql --routing-url http://localhost:4001/v2/graphql --label=team=A --feature-flag=v2
+```
+
+This will publish the new Subgraph Schema under the feature flag `products-v2`.
+Existing clients will continue to use the previous version of the Schema.
+
+To enable the feature flag for a specific request, you can set a cookie or header in your client.
+The Header Syntax is `X-WG-Feature-Flag: products.v2`.
+Multiple feature flags can be enabled by separating them with a comma: `X-WG-Feature-Flag: products.v2,employees.userAvatar`.
+
+By default, the feature flag will implicitly have a `--feature-flag-mode=replace` flag.
+This means that the Subgraph with the feature flag will replace the default Subgraph.
+
+It's also possible to use the `--feature-flag-mode=combine` flag.
+This means that the Subgraph with the feature flag will be combined with the default and other Subgraphs that have the same name.
+You cannot have Subgraphs with the same name and different feature flag modes.
+
+If two or more Subgraphs with a feature flag have the replace mode, the one published last will be used.
+
+Using the `--feature-flag-mode=combine` flag, you can publish multiple Schemas for the same Subgraph implementation,
+e.g. a base Subgraph and additional Schemas under different feature flags,
+allowing you to define how much functionality you want to expose to a client based on the feature flags that are enabled.
+
+You can remove individual feature flags from a Subgraph by using the following command:
+
+```bash
+npx wgc feature-flag remove products.v2
+```
+
+You can also publish a new Subgraph Schema and remove all existing feature flags from the Subgraph by using the following command:
+
+```bash
+npx wgc subgraph publish products --schema ../demo/subgraphs/products/products.graphql --routing-url http://localhost:4001/graphql --label=team=A --remove-feature-flags
+```
+
+When adding a feature flag to a Subgraph, we create a matrix of possible combinations of feature flags.
+This matrix is used to compute the Supergraph of all combinations of Subgraphs and feature flags.
+By default, we will run composition checks against all possible combinations of Subgraphs and feature flags to ensure that the Supergraph is valid.
+This means that future changes to any Subgraph will run composition checks against the matrix of Subgraphs and feature flags.
+
+It's possible that you want to publish a Subgraph Schema with a feature flag for an experimental feature,
```

Comment:
Kind of curious on use cases, why isn't it the default to not run composition checks for publishing feature flags? Seems like if I do want publishing to behave similarly as normal I would want to force the composition checks for feature flags rather than running composition checks for features to be the default.